projects/suirewardsme/index.js

```diff
@@ -1,29 +1,50 @@
 const sui = require("../helper/chain/sui");
 
-async function suiTVL(api) {
-    const factoryId = "0x57ecad13fb5c117a23b6e02399d3f0e539583dff324cb63761537091fe313d3d";
+const FACTORIES = [
+    {
+        id: "0x57ecad13fb5c117a23b6e02399d3f0e539583dff324cb63761537091fe313d3d",
+        name: "SuirewardsMe",
+    },
+    {
+        id: "0x4190e8f8809f81e43cc2cce98786b2f498c81281e9153c17fc4c4e382fcbc105",
+        name: "Lotus",
+    },
+];
 
-    const factory = await sui.getObject(factoryId);
-    const poolsTableId = factory.fields.pools.fields.id.id;
+async function processFactory(api, factoryId) {
+    try {
+        const factory = await sui.getObject(factoryId);
+        const poolsTableId = factory.fields.pools.fields.id.id;
 
-    const poolEntries = await sui.getDynamicFieldObjects({ parent: poolsTableId });
-    const poolObjectIds = poolEntries.map((entry) => entry.fields.value);
-    const poolObjects = await sui.getObjects(poolObjectIds);
+        const poolEntries = await sui.getDynamicFieldObjects({ parent: poolsTableId });
+        const poolObjectIds = poolEntries.map((entry) => entry.fields.value);
+        const poolObjects = await sui.getObjects(poolObjectIds);
 
-    for (const pool of poolObjects) {
-        const fields = pool.fields;
+        for (const pool of poolObjects) {
+            const { balance_a: balanceA, balance_b: balanceB } = pool.fields;
 
-        const balanceA = fields.balance_a;
-        const balanceB = fields.balance_b;
+            if (!balanceA || !balanceB || (balanceA === "0" && balanceB === "0")) continue;
 
-        if (!balanceA || !balanceB) continue;
-        if (balanceA === "0" && balanceB === "0") continue;
+            try {
+                const match = pool.type.match(/Pool<(.+?),\s*(.+?)>/);
+                if (!match) continue;
 
-        const coinA = pool.type.split("Pool<")[1].split(",")[0].trim();
-        const coinB = pool.type.split(",")[1].replace(">", "").trim();
+                const [, coinA, coinB] = match;
 
-        api.add(coinA, balanceA);
-        api.add(coinB, balanceB);
+                api.add(coinA.trim(), balanceA);
+                api.add(coinB.trim(), balanceB);
+            } catch (_) {
+                continue;
```

Comment:
this try catch could result in erratic TVL values. Our monitoring systems wont catch any errors here